util/hookable/hookable.cpp

```diff
@@ -229,18 +229,19 @@ J9HookDispatch(struct J9HookInterface **hookInterface, uintptr_t taggedEventNum,
 				}
 				OMRPORT_ACCESS_FROM_OMRPORT(commonInterface->portLib);
 				if (sampling) {
-					startTime = omrtime_current_time_millis();
+					startTime = omrtime_usec_clock(); 
 				}
 
 				function(hookInterface, eventNum, eventData, userData);
 
 				if (sampling) {
-					uint64_t timeDelta = omrtime_current_time_millis() - startTime;
+					uint64_t timeDelta = omrtime_hires_delta(startTime, omrtime_usec_clock(), OMRPORT_TIME_DELTA_IN_MICROSECONDS);
 
 					eventDump->lastHook.startTime = startTime;
 					eventDump->lastHook.callsite = record->callsite;
 					eventDump->lastHook.func_ptr = (void *)record->function;
 					eventDump->lastHook.duration = timeDelta;
+					VM_AtomicSupport::add((volatile uintptr_t *)&eventDump->totalTime, (uintptr_t)timeDelta);
```

Comment:
@LinHu2016 we don't have concerns of contention here, right? (since, frequent events are identified and sampling rate is set low)

Reply:
there might be still occasional multi threads race here , but so far no sign shows it would cause the contention.

Reply:
in line 254 if (commonInterface->threshold4Trace <= timeDelta) {
the variable **threshold4Trace** is initialized in line 144
	commonInterface->threshold4Trace = OMRHOOK_DEFAULT_THRESHOLD_IN_MILLISECONDS_WARNING_CALLBACK_ELAPSED_TIME;

the MICRO **OMRHOOK_DEFAULT_THRESHOLD_IN_MILLISECONDS_WARNING_CALLBACK_ELAPSED_TIME** is defined in omrhookable.hpp:79
since we record time in microsecond instead of millisecond, we need to update the MICRO to  
**OMRHOOK_DEFAULT_THRESHOLD_IN_MICROSECONDS_WARNING_CALLBACK_ELAPSED_TIME (100*1000)**